backend_jobs/verify_labels/pipeline_lib/firestore_database.py

```diff
@@ -123,30 +132,84 @@ def _update_label_doc(self, parent_image_ref, doc_id, label_id_list):
             database_schema.COLLECTION_IMAGES_SUBCOLLECTION_LABELS_FIELD_LABEL_IDS: label_id_list
         }, merge = True)
 
-    def _update_parent_image_labels_array(self, image_doc_ref, label_ids):
+    def _update_parent_image_labels_array_and_get_new_labels(self, image_doc_ref, label_ids):
         """ Adds the label id to the image's document if it is not already there.
 
             Args:
                 image_doc_ref: refrence to the parent image document in the database
                 label_id: the label id which needs to be added to the labels array
                 of the parent image document in the database
 
+            Yields:
+                (point key, 1).
+                point key: (precision, label id, quantized coordinates).
+                A point key for each label which was verified
+                and for each precision from 4 to 12. The quantized coordinates are
+                calculated from the parent image's hashmap.
+
+
         """
         image_doc_dict = image_doc_ref.get().to_dict()
         labels_array = []
+        parent_image_hashmap = image_doc_ref.get().to_dict()[\
+            database_schema.COLLECTION_IMAGES_FIELD_HASHMAP]
         if database_schema.COLLECTION_IMAGES_FIELD_LABELS in image_doc_dict:
             labels_array = image_doc_dict[database_schema.COLLECTION_IMAGES_FIELD_LABELS]
         for label_id in label_ids:
             if label_id not in labels_array:
-                labels_array.append(label_id) # label name instead for demo
-            image_doc_ref.update({
-                database_schema.COLLECTION_IMAGES_FIELD_LABELS: labels_array
-            })
+                labels_array.append(label_id)
+                for precision in range(4, 12):
```

Comment:
ditto, use constants. Maybe also create a function `get_point_key`

Reply:
used constants and created a method get_point_key